auto_submit/lib/requests/check_pull_request.dart

```diff
@@ -20,15 +22,26 @@ import '../server/request_handler.dart';
 class CheckPullRequest extends RequestHandler {
   CheckPullRequest({
     required Config config,
+    this.pubsub = const PubSub(),
   }) : super(config: config);
 
-  Future<Response> get(Request request) async {
-    //TODO(Kristin): Change the way to get this PR later according to the real situation, https://github.com/flutter/flutter/issues/99720
-    final String rawBody = await request.readAsString();
-    final body = json.decode(rawBody) as Map<String, dynamic>;
-    final PullRequest pullRequest = PullRequest.fromJson(body['pull_request']);
-    final RepositorySlug slug = pullRequest.base!.repo!.slug();
+  final PubSub pubsub;
+
+  Future<Response> get() async {
+    final pub.PullResponse pullResponse = await pubsub.pull(1, 'auto-submit-queue-sub');
+    final List<pub.ReceivedMessage>? receivedMessages = pullResponse.receivedMessages;
+    if (receivedMessages == null) {
+      log.info('There are no more messages available in the backlog');
+      return Response.ok('No more pull requests in the pubsub.');
+    }
+    final pub.ReceivedMessage receivedMessage = receivedMessages.first;
```

Comment:
We should make this parallelizable. The below logic can be forked into a function like `Future<void> _processMessage(ReceivedMessage message);`. Then the handler can do something like:

```dart
final List<Future<void>> futures = <Future<void>>[];
for (pub.ReceivedMessage message in pullResponse.receivedMessages) {
  futures.add(_processMessage(message));
}
await Future.wait(futures);
```

Reply:
Thank you Casey, made this parallelizable so we can deal with more than one message.